FirmataParser.h

```diff
@@ -0,0 +1,108 @@
+/*
+  Firmata.h - Firmata library v2.5.4 - 2016-10-23
+  Copyright (c) 2006-2008 Hans-Christoph Steiner.  All rights reserved.
+  Copyright (C) 2009-2016 Jeff Hoefs.  All rights reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  See file LICENSE.txt for further informations on licensing terms.
+*/
+
+#ifndef FirmataParser_h
+#define FirmataParser_h
+
+#include <cstddef>
+#include <cstdint>
+
+/* Version numbers for the protocol.  The protocol is still changing, so these
+ * version numbers are important.
+ * Query using the REPORT_VERSION message.
+ */
+#define FIRMATA_PROTOCOL_MAJOR_VERSION  2 // for non-compatible changes
+#define FIRMATA_PROTOCOL_MINOR_VERSION  5 // for backwards compatible changes
+#define FIRMATA_PROTOCOL_BUGFIX_VERSION 1 // for bugfix releases
+
+#define MAX_DATA_BYTES                  64 // max number of data bytes in incoming messages
+
+// Arduino 101 also defines SET_PIN_MODE as a macro in scss_registers.h
+#ifdef SET_PIN_MODE
```

Comment:
It would be available to anyone who includes Firmata.h, via Firmata.h's inclusion of FirmataParser.h. Currently, it sits where SET_PIN_MODE is defined and used, and you wouldn't need to worry about any funky ordering.